api_v3/services/StorageProfileService.php

```diff
@@ -410,4 +416,40 @@ protected static function getConfigVal($configMap, $configField, $defaultVal)
 		}
 		return $defaultVal;
 	}
+
+	/**
+	 * @param $workerId
+	 * @param $keysCache
+	 * @param $storageProfileId
+	 * @return int
+	 * @throws PropelException
+	 */
+	protected static function getLastFileSyncIdToLock($keysCache, $storageProfileId, $workerId)
+	{
+		$initialLastId = $keysCache->get(self::LAST_FILESYNC_ID_PREFIX . $workerId);
+		if (!$initialLastId)
+		{
+			$fileSyncThresholdCreationTime = time() - kConf::get('lastFileSyncIdCreationTimeThreshold', 'cloud_storage', 0);
```

Comment:
This value should be relative to the createdAt time of the filter, in the calling function
Since the idea is that the filter will ask for last 6 hours , but since we have no last ID , and your gap is 2Hors (for exampe) , then you should try to find 1 FS from * hours till now